app/controllers/TransferSummaryController.scala

```diff
@@ -17,44 +18,65 @@ import scala.concurrent.{ExecutionContext, Future}
 class TransferSummaryController @Inject()(val controllerComponents: SecurityComponents,
                                           val graphqlConfiguration: GraphQLConfiguration,
                                           val keycloakConfiguration: KeycloakConfiguration,
+                                          consignmentService: ConsignmentService,
                                           val consignmentExportService: ConsignmentExportService,
                                           langs: Langs)
                                          (implicit val ec: ExecutionContext) extends ValidatedActions with I18nSupport {
 
   implicit val language: Lang = langs.availables.head
-  val transferSummaryForm: Form[TransferSummaryData] = Form(
+  val transferConfirmationForm: Form[TransferConfirmationData] = Form(
     mapping(
       "openRecords" -> boolean
         .verifying(messagesApi("transferSummary.openRecords.error"), b => b),
       "transferLegalOwnership" -> boolean
         .verifying(messagesApi("transferSummary.transferLegalOwnership.error"), b => b)
-    )(TransferSummaryData.apply)(TransferSummaryData.unapply)
+    )(TransferConfirmationData.apply)(TransferConfirmationData.unapply)
   )
 
-  def transferSummary(consignmentId: UUID): Action[AnyContent] = consignmentExists(consignmentId) { implicit request: Request[AnyContent] =>
-    Ok(views.html.transferSummary(consignmentId, transferSummaryForm))
+  private def getConsignmentSummary(request: Request[AnyContent], consignmentId: UUID)
+                                   (implicit requestHeader: RequestHeader): Future[ConsignmentSummaryData] = {
+    consignmentService.getConsignmentTransferSummary(consignmentId, request.token.bearerAccessToken)
+      .map{summary =>
+        ConsignmentSummaryData(summary.series.get.code,
+          summary.transferringBody.get.name,
```

Comment:
Do you know what type of error this method would throw if you try to do this for a consignment with no consignment summary info yet?

Reply:
Yes. This is something I was speaking to Suzanne about last week. Since you need a series code and transferring body to get a consignment, that part would be fine, the only part is the file upload. Currently, if you access that page before the upload, it just says "0 files" which is ok, but we probably wouldn't want them to continue on and click "transfer your records". Accessing pages before you are supposed to is something that Pia is looking into.